internal/services/network/network_interface_resource.go

```diff
@@ -364,142 +363,146 @@ func resourceNetworkInterfaceUpdate(d *pluginsdk.ResourceData, meta interface{})
 		// then map the fields managed in other resources back
 		ipConfigs = mapFieldsToNetworkInterface(ipConfigs, info)
 
-		update.InterfacePropertiesFormat.IPConfigurations = ipConfigs
+		update.Properties.IPConfigurations = ipConfigs
 	} else {
-		update.InterfacePropertiesFormat.IPConfigurations = existing.InterfacePropertiesFormat.IPConfigurations
+		update.Properties.IPConfigurations = existing.Model.Properties.IPConfigurations
 	}
 
 	if d.HasChange("tags") {
 		tagsRaw := d.Get("tags").(map[string]interface{})
-		update.Tags = tags.Expand(tagsRaw)
+		update.Tags = utils.ExpandPtrMapStringString(tagsRaw)
```

Comment:
Apply that same `tag` logic from the datasource here

Reply:
Code updated.